src/test/java/org/prebid/server/bidder/improvedigital/ImprovedigitalBidderTest.java

```diff
@@ -67,6 +70,52 @@ public void makeHttpRequestsShouldNotModifyIncomingRequest() {
                 .containsExactly(bidRequest);
     }
 
+    @Test
+    public void makeHttpRequestsShouldReturnErrorIfImpExtCouldNotBeParsed() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode())))
+                        .build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = improvedigitalBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors().get(0).getMessage()).startsWith("Cannot deserialize instance");
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorOnMissingPlacementId() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createObjectNode())))
+                        .build()))
+                .id("request_id")
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = improvedigitalBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors().get(0).getMessage()).startsWith("No placementId provided");
```

Comment:
Can be replaced with
`assertThat(result.getErrors()).containsExactly(BidderError.badInput("No placementId provided"))`, also pls check similar occurrences